diffcp/cone_program.py

```diff
@@ -510,10 +515,10 @@ def solve_internal(A, b, c, cone_dict, solve_method=None,
     return result
 
 def solve_and_derivative_internal(A, b, c, cone_dict, solve_method=None,
-        warm_start=None, mode='lsqr', raise_on_error=True, **kwargs):
-    if mode not in ["dense", "lsqr", "lsmr"]:
+        warm_start=None, mode='lsqr', raise_on_error=True, P=None, **kwargs):
+    if mode not in ["dense", "lsqr", "lsmr", "lpgd", "lpgd_right", "lpgd_left"]:
         raise ValueError("Unsupported mode {}; the supported modes are "
-                         "'dense', 'lsqr' and 'lsmr'".format(mode))
+                         "'dense', 'lsqr', 'lsmr', 'lpgd', 'lpgd_right' and 'lpgd_left'".format(mode))
     if np.isnan(A.data).any():
```

Comment:
Shouldn't we check here if `P is None` for the `dense, lsqr, lsmr` cases? They don't support quadratic objectives in my understanding.

Reply:
Yes that makes sense, I will add this.